File: 백준/Bronze/1110. 더하기 사이클/더하기 사이클.java
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;

public class Main {
	static BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
	
	public static void main(String[] args) throws NumberFormatException, IOException {
		int N = Integer.parseInt(br.readLine());
		
		if (N / 10 < 1) {
			N *= 10;
		}
		int cycle = 0;
		int newN = N;
		while (true) {
			newN = (newN % 10 + (int) (newN / 10)) % 10 + (newN % 10) * 10;
			cycle++;
			if (newN == N) {
				break;
			}
		}
		
		System.out.println(cycle);
	} // end of main
} // end of class

File: 백준/Bronze/1110. 더하기 사이클/더하기 사이클.py
import sys

input = lambda: sys.stdin.readline().strip()

N = int(input())

if N < 10:
    N *= 10

new_N = N
cycle = 0
while True:
    new_N = (new_N % 10 * 10) + (new_N % 10 + new_N // 10) % 10
    cycle += 1
    if new_N == N:
        break

print(cycle)